File: app/src/main/java/com/androidnerds/doordash/data/local/IStoreFeedLocalDataSource.java
package com.androidnerds.doordash.data.local;

import com.androidnerds.doordash.data.local.entity.StoreFeedEntity;

import io.reactivex.rxjava3.core.Maybe;
import io.reactivex.rxjava3.core.Single;


public interface IStoreFeedLocalDataSource {
    Maybe<StoreFeedEntity> getStoreFeed(double latitude, double longitude, int limit, long offset);
    Single<Integer> getStoreFeedCount();
}

File: app/src/main/java/com/androidnerds/doordash/storefeed/data/local/IStoreFeedLocalDataSource.kt
package com.androidnerds.doordash.storefeed.data.local

import com.androidnerds.doordash.storefeed.domain.Location
import com.androidnerds.doordash.storefeed.domain.StoreFeed
import io.reactivex.Flowable

interface IStoreFeedLocalDataSource {

    fun getStoreFeed(location: Location, limit: Int, offset: Int) : Flowable<StoreFeed>
}